numpy/random/mtrand.pyx

```diff
@@ -3728,6 +3768,11 @@ cdef class RandomState:
             raise ValueError("cov must be 2 dimensional and square")
         if mean.shape[0] != cov.shape[0]:
             raise ValueError("mean and cov must have same length")
+        # raise exception if cov is not symmetric
+        if mode != 'factor':
+            symmetric = np.allclose(cov, cov.T, rtol=tol, atol=tol)
+            if not symmetric:
+                raise ValueError("covariance is not symmetric.")
```

Comment:
Do we want to add this relatively expensive check by default? Perhaps it should be behind the `check_valid` flag?